### PR TITLE
costo: new package

### PR DESCRIPTION
**Pull request by tech-91**

### CODE REVIEW

**Comment by tech-91** [comment]
Is it possible to merge this PR?

**Review by bernhardkaindl — COMMENTED** [review]
@tech-91 
> Is it possible to merge this PR?

I would have wanted to check if the packages are buildable, but the dependency `py-mgmetis` still fails to build:
```py
==> Installing py-mgmetis-master-vkvlqarpuo3earxqvbjdgbifvn524lhz [124/134]
==> py-mgmetis: Executing phase: 'install'
==> [2024-10-24-05:27:59.894587] 'opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/python-venv-1.0-mg2gbcwegmfesk2372yweq4p6ntbx555/bin/python3' '-m' 'pip' '-vvv' '--no-input' '--no-cache-dir' '--disable-pip-version-check' 'install' '--no-deps' '--ignore-installed' '--no-build-isolation' '--no-warn-script-location' '--no-index' '--prefix=opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/py-mgmetis-master-vkvlqarpuo3earxqvbjdgbifvn524lhz' '.'
Using pip 23.1.2 from opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/py-pip-23.1.2-kiyxspu7orsdqdxrezghvbnfx43s43g3/lib/python3.11/site-packages/pip (python 3.11)
Non-user install due to --prefix or --target option
Ignoring indexes: https://pypi.org/simple
Created temporary directory: /tmp/pip-build-tracker-8v0fif_o
Initialized build tracking at /tmp/pip-build-tracker-8v0fif_o
Created build tracker: /tmp/pip-build-tracker-8v0fif_o
Entered build tracker: /tmp/pip-build-tracker-8v0fif_o
Created temporary directory: /tmp/pip-install-5sxnpcb8
Created temporary directory: /tmp/pip-ephem-wheel-cache-0n3ew2zr
Processing .stage/spack-stage-py-mgmetis-master-vkvlqarpuo3earxqvbjdgbifvn524lhz/spack-src
  Added file://.stage/spack-stage-py-mgmetis-master-vkvlqarpuo3earxqvbjdgbifvn524lhz/spack-src to build tracker '/tmp/pip-build-tracker-8v0fif_o'
  Running setup.py (path:.stage/spack-stage-py-mgmetis-master-vkvlqarpuo3earxqvbjdgbifvn524lhz/spack-src/setup.py) egg_info for package from file://.stage/spack-stage-py-mgmetis-master-vkvlqarpuo3earxqvbjdgbifvn524lhz/spack-src
  Created temporary directory: /tmp/pip-pip-egg-info-7mhwndgt
  Preparing metadata (setup.py): started
  Running command python setup.py egg_info
  Traceback (most recent call last):
    File "<string>", line 2, in <module>
    File "<pip-setuptools-caller>", line 34, in <module>
    File ".stage/spack-stage-py-mgmetis-master-vkvlqarpuo3earxqvbjdgbifvn524lhz/spack-src/setup.py", line 5, in <module>
      import conf
    File ".stage/spack-stage-py-mgmetis-master-vkvlqarpuo3earxqvbjdgbifvn524lhz/spack-src/conf.py", line 74, in <module>
      os.environ["CC"] = mpi4py.get_config()["mpicc"]
                         ~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  KeyError: 'mpicc'
``` 
Does `py-mgmetis` need a `depends_on("mpi")` or so?

Can you build it from an empty spack repository without any cached or external packages?

Can you check or explain what needs to be done for a successful build?

If found that `py-mpi4py` has recently been updated to `@4.0.0`, could this be the cause?
Update: Yes, it seems to be.

File: var/spack/repos/builtin/packages/costo/package.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    depends_on("mpi", type=all)
+    depends_on("python@3.0:", type=all)
+
+    depends_on("py-non-regression-test-tools", type="build")
+    depends_on("py-pyvista", type="run")
+    depends_on("py-numpy", type=("build", "link", "run"))
+    depends_on("py-mpi4py", type="run")
+    depends_on("py-scipy", type="run")
+    depends_on("py-mgmetis", type="run")
+    depends_on("py-colorama", type="run")
```
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/costo/package.py]: These dependencies are missing versions, and it looks like there are some versions that are tested:
https://gitlab.com/Te_ch/costo/-/commit/24bd6a0edb461d2df93e2b3a9c663d346f2b2193#28aee9c390897ec2b7fc8986d77695da15dd8c42_0_34

Please add them, and it says it needs `python@3.10:`, mot `@3.0:`.

Why does your project not even have a README that tells about it?

File: var/spack/repos/builtin/packages/costo/package.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Costo(CMakePackage):
+    """Cosimulation Tools"""
```
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/costo/package.py]: "Cosimulation tools" is quite short, what is it, and what is it used for?

**Review by bernhardkaindl — CHANGES_REQUESTED** [review]

**Review by bernhardkaindl — CHANGES_REQUESTED** [review]
I manually cloned the project, installed `mpich` and ran cmake:
```py
cmake -B build
-- Could NOT find Python3 (missing: Python3_NumPy_INCLUDE_DIRS NumPy) (found version "3.10.12")
-- GIT_EXE: /usr/bin/git
-- GIT_HASH: f9e3ee7
-- GIT_DIFF:
-- GIT_TAG:
-- GIT_BRANCH: HEAD
-- PROJECT_VERSION: 0.0.4
CMake Error: File costo/python/costo/__init__.py.in does not exist.
CMake Error at CMakeLists.txt:84 (configure_file):
  configure_file Problem configuring file


Defaulting to user installation because normal site-packages is not writeable
Processing ./python
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [48 lines of output]
      Traceback (most recent call last):
        File "/usr/lib/python3/dist-packages/setuptools/config.py", line 419, in _parse_attr
          return getattr(StaticModule(module_name), attr_name)
        File "/usr/lib/python3/dist-packages/setuptools/config.py", line 27, in __init__
          with open(spec.origin) as strm:
      AttributeError: 'NoneType' object has no attribute 'origin'
```
This is the same error that I also get in spack now.
It looks like `costo/python/costo/__init__.py.in` is missing and maybe more is broken (it looks like at least).

**Comment by tech-91** [comment]
Hy,

Thanks for looking the error.
In fact, the file costo/python/costo/__init__.py.in should be found in
costo/python/src/_init__.py.in which mean the path seems broken.
I'll check this this morning and keep you informed.



Le jeu. 24 oct. 2024 à 09:39, Bernhard Kaindl ***@***.***> a
écrit :

File: var/spack/repos/builtin/packages/costo/package.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    maintainers("tech-91")
+
+    license("LGPL-3.0-or-later")
+
+    version(
+        "0.0.4", tag="v0.0.4", commit="f9e3ee774a33cc3af566e338d542d7dd0ff36cea", preferred=True
+    )
+    version("develop", branch="devel")
+    version("master", branch="main", deprecated=True)
```
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/costo/package.py]: There is no `master`: https://gitlab.com/Te_ch/costo/-/branches
There is a `develop`, but the only difference to 0.0.4 is that it's version is 0.0.0. Skip it.
There is a `main` but its only difference is in the CI.

As there is no documentation at all that I could find, I am skeptical what the value of adding it to the main `spack` repo would be.

@tech-91, you you have a good explanation why this is needed as a spack package?

At least a small semblance of some doc what it even is would be required, IMHO.

**Review by bernhardkaindl — CHANGES_REQUESTED** [review]

**Comment by bernhardkaindl** [comment]
> Thanks for looking the error.
> In fact, the file `costo/python/costo/__init__.py.in` should be found in
> `costo/python/__init__.py.in` which mean yhe path seems broken.
> I'll check this this morning and keep you informed.

There is one in `python/src/costo/__init__.py.in`:
```py
python/costo/__init__.py.in      - does not exist, but it expected by cmake
python/src/costo/__init__.py.in  - exists.
```

**Comment by tech-91** [comment]
I juste made a push with the corresponding fixe on develop & main branches.


Le jeu. 24 oct. 2024 à 09:54, Bernhard Kaindl ***@***.***> a
écrit :

**Comment by tech-91** [comment]
I just made a push with the corresponding fix on develop & main branches.
This project is under development, this is why the doc still remain almost empty.
Also I need it on spack in order to deploy it on some air gap machines.



Le jeu. 24 oct. 2024 à 09:55, Chantrait Teddy ***@***.***> a
écrit :

File: var/spack/repos/builtin/packages/costo/package.py (hunk 1 above)
Comment (tech-91) [reply var/spack/repos/builtin/packages/costo/package.py]: Done on commit https://github.com/spack/spack/pull/47121/commits/4b1c816215127a91040dadd0078525e14dcaebd5

**Review by bendamasio — CHANGES_REQUESTED** [review]
<details><summary>Hello @tech-91! I encountered an install failure of costo+tests@=0.0.5. Could you check it?</summary><br>

Hello @tech-91,

I am terribly sorry, but my attempt on the installation of `costo+tests@=0.0.5` failed in an automated build.

 But I've got a detailed report for you that may allow you to fix it based on the knowledge that you have about these recipes.



This command failed: `bin/spack install -v --fail-fast costo+tests@=0.0.5`
<details><summary>Click here to drill down to the logs from building the failed spec: costo+tests@=0.0.5</summary>

### `costo+tests@=0.0.5`:
<details><summary><b>Click here to expand the full build log</b> for costo+tests@=0.0.5</summary>

```py
==> costo: Executing phase: 'cmake'
==> [2024-11-01-05:22:01.595749] '$PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/cmake-3.30.5-suh3nit66ur4vbikvcxn6e5kvuxco5jc/bin/cmake' '-G' 'Unix Makefiles'
-- The CXX compiler identification is GNU 13.1.0
-- The C compiler identification is GNU 13.1.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: $PWD/lib/spack/env/gcc/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: $PWD/lib/spack/env/gcc/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Found MPI_C: $PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/openmpi-5.0.5-22mbihdnstnzindmy2ix545q2z6itgzg/lib/libmpi.so (found version "3.1")
-- Found MPI_CXX: $PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/openmpi-5.0.5-22mbihdnstnzindmy2ix545q2z6itgzg/lib/libmpi.so (found version "3.1")
-- Found MPI: TRUE (found version "3.1")
-- Found Python3: $PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/python-3.10.14-ofmmwwhrgu76ks5gigxvqk4rf5anndnr/bin/python3.10 (found version "3.10.14") found components: Interpreter Development NumPy Development.Module Development.Embed
-- GIT_EXE: /usr/bin/git
-- GIT_HASH: N/A
-- GIT_DIFF: 
-- GIT_TAG: N/A
-- GIT_BRANCH: N/A
-- PROJECT_VERSION: 0.0.4
CMake Error at tests/CMakeLists.txt:27 (find_package):
  By not providing "FindCosto.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Costo", but
  CMake did not find one.
  Could not find a package configuration file provided by "Costo" with any of
  the following names:
    CostoConfig.cmake
    costo-config.cmake
  Add the installation prefix of "Costo" to CMAKE_PREFIX_PATH or set
  "Costo_DIR" to a directory containing one of the above files.  If "Costo"
  provides a separate development package or SDK, be sure it has been
  installed.
-- Configuring incomplete, errors occurred!

```
</details>

```py
-- Configuring incomplete, errors occurred!
==> Error: ProcessError: Command exited with status 1:
    opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/cmake-3.30.5-suh3nit66ur4vbikvcxn6e5kvuxco5jc/bin/cmake -G Unix Makefiles -DCMAKE_PREFIX_PATH:STRING=opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/py-non-regression-test-tools-1.1.2-o2kkithqx37u4qytqsgzf7zqtalxjk4b;opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/py-numpy-1.26.4-ct7imsszwp7gfaxzoxyqr6q6t2htx7vh;opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/openblas-0.3.28-xrdrjso7wej7ivc4g2jkplsl3o5273jh;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/py-pip-23.1.2-wnvjstj7jqxspizkq4wtyrx54h5ihwp5;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/python-3.10.14-ofmmwwhrgu76ks5gigxvqk4rf5anndnr;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/util-linux-uuid-2.40.2-bqq7v2o43bwegp76qc2oce2xmbcwrduq;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/sqlite-3.46.0-nduy7gea62eelqexs6o4xyl2rasruzls;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libffi-3.4.6-xqu3yeo5z7azukj7nkqfvjzmixhp44dy;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/expat-2.6.3-wch5dt2lba7wfls3liws7iwqdfkral5y;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libbsd-0.12.2-cp37gsuedh4xwswccbnh7cyjtdr6fy4j;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libmd-1.0.4-mvecdr3afwkcfeolgt7qsbt5uf4pwecz;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/openmpi-5.0.5-22mbihdnstnzindmy2ix545q2z6itgzg;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/pmix-5.0.3-ken7b7ozlb4rqo4snvj26p6ht4kbytld;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libxcrypt-4.4.35-bh4rhzp7kqgua6zjmgqw5f2dkjlzsvwf;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/gettext-0.22.5-6jf4uw2w6roexfqdg7n7k5kzpeosswby;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/numactl-2.0.18-sn7n3t6w7qfrloozixy46okdwjjy6zyw;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libevent-2.1.12-y76shbggve3qimgoyedvrjyln4bur4nt;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/hwloc-2.11.1-jaot5abqrnvjje7flflrgik42uv267zk;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libxml2-2.10.3-hwvd3j4rs3gm2hxonrv466wwp2dwwawt;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/xz-5.4.6-mturqa5up3ztik5n5d2vrpqwkv5smix7;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libpciaccess-0.17-lhqqfllyhxnw3neo5c3hswjtfc4eqmq3;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/cmake-3.30.5-suh3nit66ur4vbikvcxn6e5kvuxco5jc;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/openssl-3.3.1-7yojqoveilpwme7bhspzcwag635qghiw;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/zlib-ng-2.2.1-c5rauqdmsth563m56xk2i26gh7croe44;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/gdbm-1.23-ywkyzdbsxd2u55gyrjuqkvsidcwu5ovo;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/readline-8.2-hifil7lrxuysqa45pht5cn26tbb37lv7;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/ncurses-6.5-2yuat3jia57sfgfoc7fdsl7arajm7tfe;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/bzip2-1.0.8-qqzuq3npoxz37mviu3f2r6xnydsy6wca;$PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/gmake-4.4.1-krhvoy5tc5wo3p2bpim66lm35gbmykuj;opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/gcc-runtime-13.1.0-25qo3ld7klulv5juopmjnk73qte743xz -DWITH_PYTHON_MODULE:STRING=ON -DWITH_SHARED_LIBS:BOOL=ON -DWITH_TESTS:BOOL=ON .stage/spack-stage-costo-0.0.5-ughaxpse7o27oby7o5phymeejjn55vcs/spack-src

     24    -- PROJECT_VERSION: 0.0.4
  >> 25    CMake Error at tests/CMakeLists.txt:27 (find_package):
     26      By not providing "FindCosto.cmake" in CMAKE_MODULE_PATH this project has
     27      asked CMake to find a package configuration file provided by "Costo", but
```

</details>


</details>

**Review by bendamasio — CHANGES_REQUESTED** [review]
<details><summary>Hello @tech-91! I encountered an install failure of costo+tests@=develop. Could you check it?</summary><br>

Hello @tech-91,

I am terribly sorry, but my attempt on the installation of `costo+tests@=develop` failed in an automated build.

 But I've got a detailed report for you that may allow you to fix it based on the knowledge that you have about these recipes.



This command failed: `bin/spack install -v --fail-fast costo+tests@=develop`
<details><summary>Click here to drill down to the logs from building the failed spec: costo+tests@=develop</summary>

### `costo+tests@=develop`:
<details><summary><b>Click here to expand the full build log</b> for costo+tests@=develop</summary>

```py
==> costo: Executing phase: 'cmake'
==> [2024-11-01-05:24:06.885314] '$PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/cmake-3.30.5-suh3nit66ur4vbikvcxn6e5kvuxco5jc/bin/cmake' '-G' 'Unix Makefiles'
-- The CXX compiler identification is GNU 13.1.0
-- The C compiler identification is GNU 13.1.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: $PWD/lib/spack/env/gcc/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: $PWD/lib/spack/env/gcc/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Found MPI_C: $PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/openmpi-5.0.5-22mbihdnstnzindmy2ix545q2z6itgzg/lib/libmpi.so (found version "3.1")
-- Found MPI_CXX: $PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/openmpi-5.0.5-22mbihdnstnzindmy2ix545q2z6itgzg/lib/libmpi.so (found version "3.1")
-- Found MPI: TRUE (found version "3.1")
-- Found Python3: $PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/python-3.10.14-ofmmwwhrgu76ks5gigxvqk4rf5anndnr/bin/python3.10 (found version "3.10.14") found components: Interpreter Development NumPy Development.Module Development.Embed
-- GIT_EXE: /usr/bin/git
-- GIT_HASH: 6660d69
-- GIT_DIFF: 
-- GIT_TAG: 
-- GIT_BRANCH: devel
-- PROJECT_VERSION: 0.0.4
CMake Error at tests/CMakeLists.txt:27 (find_package):
  By not providing "FindCosto.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Costo", but
  CMake did not find one.
  Could not find a package configuration file provided by "Costo" with any of
  the following names:
    CostoConfig.cmake
    costo-config.cmake
  Add the installation prefix of "Costo" to CMAKE_PREFIX_PATH or set
  "Costo_DIR" to a directory containing one of the above files.  If "Costo"
  provides a separate development package or SDK, be sure it has been
  installed.
-- Configuring incomplete, errors occurred!

```
</details>

```py
-- Configuring incomplete, errors occurred!
==> Error: ProcessError: Command exited with status 1:
    opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/cmake-3.30.5-suh3nit66ur4vbikvcxn6e5kvuxco5jc/bin/cmake -G Unix Makefiles -DCMAKE_PREFIX_PATH:STRING=opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/py-non-regression-test-tools-1.1.2-o2kkithqx37u4qytqsgzf7zqtalxjk4b;opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/py-numpy-1.26.4-ct7imsszwp7gfaxzoxyqr6q6t2htx7vh;opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/openblas-0.3.28-xrdrjso7wej7ivc4g2jkplsl3o5273jh;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/py-pip-23.1.2-wnvjstj7jqxspizkq4wtyrx54h5ihwp5;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/python-3.10.14-ofmmwwhrgu76ks5gigxvqk4rf5anndnr;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/util-linux-uuid-2.40.2-bqq7v2o43bwegp76qc2oce2xmbcwrduq;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/sqlite-3.46.0-nduy7gea62eelqexs6o4xyl2rasruzls;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libffi-3.4.6-xqu3yeo5z7azukj7nkqfvjzmixhp44dy;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/expat-2.6.3-wch5dt2lba7wfls3liws7iwqdfkral5y;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libbsd-0.12.2-cp37gsuedh4xwswccbnh7cyjtdr6fy4j;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libmd-1.0.4-mvecdr3afwkcfeolgt7qsbt5uf4pwecz;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/openmpi-5.0.5-22mbihdnstnzindmy2ix545q2z6itgzg;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/pmix-5.0.3-ken7b7ozlb4rqo4snvj26p6ht4kbytld;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libxcrypt-4.4.35-bh4rhzp7kqgua6zjmgqw5f2dkjlzsvwf;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/gettext-0.22.5-6jf4uw2w6roexfqdg7n7k5kzpeosswby;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/numactl-2.0.18-sn7n3t6w7qfrloozixy46okdwjjy6zyw;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libevent-2.1.12-y76shbggve3qimgoyedvrjyln4bur4nt;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/hwloc-2.11.1-jaot5abqrnvjje7flflrgik42uv267zk;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libxml2-2.10.3-hwvd3j4rs3gm2hxonrv466wwp2dwwawt;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/xz-5.4.6-mturqa5up3ztik5n5d2vrpqwkv5smix7;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libpciaccess-0.17-lhqqfllyhxnw3neo5c3hswjtfc4eqmq3;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/cmake-3.30.5-suh3nit66ur4vbikvcxn6e5kvuxco5jc;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/openssl-3.3.1-7yojqoveilpwme7bhspzcwag635qghiw;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/zlib-ng-2.2.1-c5rauqdmsth563m56xk2i26gh7croe44;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/gdbm-1.23-ywkyzdbsxd2u55gyrjuqkvsidcwu5ovo;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/readline-8.2-hifil7lrxuysqa45pht5cn26tbb37lv7;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/ncurses-6.5-2yuat3jia57sfgfoc7fdsl7arajm7tfe;opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/bzip2-1.0.8-qqzuq3npoxz37mviu3f2r6xnydsy6wca;$PWD/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/gmake-4.4.1-krhvoy5tc5wo3p2bpim66lm35gbmykuj;opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/gcc-runtime-13.1.0-25qo3ld7klulv5juopmjnk73qte743xz  -DWITH_PYTHON_MODULE:STRING=ON -DWITH_SHARED_LIBS:BOOL=ON -DWITH_TESTS:BOOL=ON .stage/spack-stage-costo-develop-d3657vr3og74okmwtjjogdqvqcnhtzyy/spack-src

     24    -- PROJECT_VERSION: 0.0.4
  >> 25    CMake Error at tests/CMakeLists.txt:27 (find_package):
     26      By not providing "FindCosto.cmake" in CMAKE_MODULE_PATH this project has
     27      asked CMake to find a package configuration file provided by "Costo", but
```

</details>


</details>

**Review by bendamasio — CHANGES_REQUESTED** [review]
(obsolete)

**Review by bernhardkaindl — CHANGES_REQUESTED** [review]
The remaining issues that x64-Linux-Builder reports should also be fixed, I guess.

**Review by bendamasio — APPROVED** [review]
`spack install costo@0.5.0` was successful on x86-Linux

**Review by bernhardkaindl — APPROVED** [review]
> spack install costo@0.5.0 was successful on x86-Linux

My review comments appear to be addressed and the build was fixed.

**Pull request merged**